src/Silex/ArgumentResolver.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Silex framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Silex;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentResolverInterface;
+
+/**
+ * HttpKernel Argument Resolver for Silex.
+ *
+ * @author Romain Neutron <imprec@gmail.com>
+ */
+class ArgumentResolver implements ArgumentResolverInterface
+{
+    private $argumentResolver;
+    private $app;
+
+    public function __construct(ArgumentResolverInterface $argumentResolver)
+    {
+        $this->argumentResolver = $argumentResolver;
+    }
+
+    public function setApplication(Application $app)
+    {
+        $this->app = $app;
+    }
+
+    public function getArguments(Request $request, $controller)
+    {
+        if (null === $this->app) {
+            throw new \LogicException('You should inject Silex Application prior to call ArgumentResolver::getArguments');
+        }
+
+        if (is_array($controller)) {
+            $r = new \ReflectionMethod($controller[0], $controller[1]);
+        } elseif (is_object($controller) && !$controller instanceof \Closure) {
+            $r = new \ReflectionObject($controller);
+            $r = $r->getMethod('__invoke');
+        } else {
+            $r = new \ReflectionFunction($controller);
+        }
+
+        foreach ($r->getParameters() as $param) {
+            if ($param->getClass() && $param->getClass()->isInstance($this->app)) {
+                $request->attributes->set($param->getName(), $this->app);
```

Comment:
-1 for that. Injecting the app in the request attribute was a hacky way (the same than ParamConverters in SensioFrameworkExtraBundle). You should use the Symfony 3.1 extension points instead, by using a ArgumentValueResolver for the application